SNLS_NLSPowell.h

```diff
@@ -0,0 +1,729 @@
+    {
+        bool retval = this->m_crj.computeRJ(r, J, m_x);
+        // Add the finite difference portion of things back in here at a later point        
+        return retval;  
+    }
+
+    private:
+
+    __snls_hdev__
+    inline
+    SNLSStatus_t solveInit(double* const residual)
+    {
+        m_jevals = 0;
+        m_fevals = 1; 
+        m_status = snls::unConverged;
+        // at m_x
+        bool rjSuccess = this->computeRJ(residual, nullptr);
+        if ( !(rjSuccess) ) {
+            m_status = initEvalFailure;
+            return m_status;
+        }
+        
+        
+        m_res = snls::linalg::norm<m_nDim>(residual);
+        
+        // initialize iteration counter and monitors
+        m_nIters = 1;
+        m_ncsuc = 0;
+        m_ncfail = 0;
+        m_nslow1 = 0;
+        m_nslow2 = 0;
+        return m_status;
+    }
+
+    __snls_hdev__
+    inline
+    SNLSStatus_t solveStep(double* const residual,
+                           double* const Jacobian,
+                           double* const Q,
+                           double* const diag,
+                           double* const qtf,
+                           double* const work_arr1,
+                           double* const work_arr2,
+                           double* const work_arr3)
+    {
+        const double epsmch = snls::dbl_epsmch;
+        bool jeval = true;
+        
+        // at m_x
+        // We currently don't check on the success of this...
+        // and we should at this back in at some point
+        // However in the mean time, this is to quite the compiler
+        // bool rjSuccess = 
+        this->computeRJ(residual, Jacobian);
+        // Need to think about what to do if this fails as things
+        // are handled differently from the TrDL solver.
+        // if ( !(rjSuccess) ) {
+        //     m_status = initEvalFailure;
+        //     return m_status;
+        // }
+        m_jevals += 1;
+
+
+        snls::linalg::normColumn<m_nDim, m_nDim>(Jacobian, work_arr2);
+
+        if(m_nIters == 1) {
+            if (scaleDiag) {
+                for (int i = 0; i < m_nDim; i++) {
+                    if (work_arr2[i] != 0.0) {
+                        diag[i] = work_arr2[i];
+                    } else {
+                        diag[i] = 1.0;
+                    }
+                }
+            }
+
+            m_xnorm = snls::linalg::norm<m_nDim>(m_x);
+            m_delta = m_factor * m_xnorm;
+            if (m_delta == 0.0) {
+                m_delta = m_factor; 
+            }
+        }
+
+        if (scaleDiag) {
+            for (int i = 0; i < m_nDim; i++) {
+                diag[i] = (diag[i] > work_arr2[i]) ? diag[i] : work_arr2[i];
+            }
+        }
+
+        // Jacobian is our R matrix and Q
+        snls::houseHolderQR<m_nDim, m_nDim>(Jacobian, Q, work_arr1, work_arr2, work_arr3);
+
+        snls::linalg::matTVecMult<m_nDim, m_nDim>(Q, residual, qtf);
+
+        // std::cout << "R" << std::endl;
```

Comment:
Either remove this dead code or put it in a debug ifdef?

Reply:
Missed that when pushing the commits up. I'll remove that statement as it was used while debugging.